### ISSUE TITLE
"Access is allowed from Event Dispatch Thread (EDT) only" after upgrade to Idea 2023.3.1

### ISSUE THREAD

**Issue by Igor-Mukhin** [jshiell/checkstyle-idea#626]
After upgrade to Idea 2023.3.1 after some actions in the IDE (like a Git commit), I am getting a plugin error message:

```
java.lang.Throwable: Access is allowed from Event Dispatch Thread (EDT) only
	at com.intellij.openapi.vcs.changes.ui.ChangesTree.getInclusionModel(ChangesTree.java:549)
	at com.intellij.openapi.vcs.changes.ui.ChangesTree.getIncludedSet(ChangesTree.java:617)
	at com.intellij.openapi.vcs.changes.ui.VcsTreeModelData$IncludedUnderData.<init>(VcsTreeModelData.java:255)
	at com.intellij.openapi.vcs.changes.ui.VcsTreeModelData.included(VcsTreeModelData.java:54)
	at com.intellij.vcs.commit.ChangesViewCommitPanel.getIncludedChanges(ChangesViewCommitPanel.kt:191)
	at com.intellij.vcs.commit.AbstractCommitWorkflowHandlerKt.getIncludedPaths(AbstractCommitWorkflowHandler.kt:42)
	at com.intellij.vcs.commit.CommitProjectPanelAdapter.getVirtualFiles(CommitProjectPanelAdapter.kt:29)
	at org.infernus.idea.checkstyle.handlers.ScanFilesBeforeCheckinHandler$2.run(ScanFilesBeforeCheckinHandler.java:88)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:434)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:132)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressSynchronously$8(CoreProgressManager.java:518)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$new$0(ProgressRunner.java:84)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:248)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:73)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:192)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:609)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:73)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:248)
	at com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:498)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:81)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:81)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:86)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:81)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:464)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

**Comment by jshiell** [comment]
Thanks for the report. Most likely cause is that the latest API requires this particular API to be called on the event thread; I'll have a look hopefully later this weekend.

**Comment by jshiell** [comment]
[2023.3 brings threading model changes](https://plugins.jetbrains.com/docs/intellij/general-threading-rules.html)gst, so we'll be lucky if it's just this one 😢 

But this one was easy to reproduce and should be fixed in 5.85.1. I say should as I don't have much time for testing today, but it certainly seemed to solve the case I saw 🤞

**Comment by jshiell** [comment]
Smashing - thanks for letting me know!
